libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2508,29 +2507,56 @@ string YulUtilFunctions::revertReasonIfDebug(RevertStrings revertStrings, string
 				mstore(add(reasonPos, <offset>), <wordValue>)
 			</word>
 			revert(0, add(reasonPos, <end>))
-		})");
-		templ("sig", (u256(util::FixedHash<4>::Arith(util::FixedHash<4>(util::keccak256("Error(string)")))) << (256 - 32)).str());
+		<!debugAndMessage>
+			revert(0, 0)
+		</debugAndMessage>
+	})");
+	bool debugAndMessage = _revertStrings >= RevertStrings::Debug && !_message.empty();
```

Comment:
I think this is more readable with extra parentheses that force you to read the condition first:
```suggestion
	bool debugAndMessage = (_revertStrings >= RevertStrings::Debug && !_message.empty());
```